src/js/components/CollapsibleErrorMessage.js

```diff
@@ -0,0 +1,227 @@
+import Icon from './Icon';
+
+import React from 'react';
+import classNames from 'classnames';
+
+/**
+ * Methods to bind in 'this' context
+ */
+const METHODS_TO_BIND = [
+  'toggleExpanded'
+];
+
+/**
+ * An error bar component commonly used in the ServiceForm and the JobsForm
+ * that contains collapsible error messages.
+ *
+ * @example <caption>How to use the CollapsibleErrorMessage</caption>
+ *   <CollapsibleErrorMessage
+ *
+ *        className="error-for-modal"
+ *
+ *        details={[ 'an', 'array', 'of', 'detailed errors' ]}
+ *        message="The message to display"
+ *
+ *        onToggle={function(isToggled) {
+ *          // Do something
+ *        }}
+ *
+ *      />
+ */
+class CollapsibleErrorMessage extends React.Component {
+
+  /**
+   * Initialize superclass
+   */
+  constructor() {
+    super(...arguments);
+
+    // Initial state
+    this.state = {
+      expanded: false
+    };
+
+    // Bind methods in context
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+
+  }
+
+  /**
+   * Set the default expandedg from the properties
+   */
+  componentWillMount() {
+    this.setState({
+      expanded: this.props.expanded
+    });
+  }
+
+  /**
+   * Toggle the expanded state of the collapsible error bar
+   */
+  toggleExpanded() {
+    let expanded = !this.state.expanded;
+    this.setState({expanded});
+
+    // Call onToggle callback
+    this.props.onToggle(expanded);
+  }
+
+  /**
+   * [Fragment] The show more/less link
+   *
+   * @returns {React.Component} - The rendered content
+   */
+  getShowDetailsLink() {
+    let {message, details} = this.props;
+    let isDetailed = message && details && details.length;
+
+    // Check if we must not show the detail link at all
+    if (!isDetailed) {
+      return null;
+    }
```

Comment:
I'd argue there is no need for the `isDetailed` variable, we could just 
```
let {message, details} = this.props;
if (message != null && details != null && details.length) {
  return null;
}
```
but that's really minor knittin' 😛